### DISCUSSION TITLE
Non-Zero Exit Code When Running Prefect Deployment

### DISCUSSION THREAD

**Discussion by NembotJules** [PrefectHQ/prefect#16114] [General] [answered]
### Bug summary

I consistently encounter the following error when running a Prefect deployment:

```
Flow run infrastructure exited with non-zero status code:
Exited with non-zero code. (Error Code: 1)
This may be caused by attempting to run an image with a misspecified platform or architecture.
```
 

Here is the code I am using for the deployment:

```python
if __name__ == "__main__":
    main_flow.from_source(
        source="https://github.com/NembotJules/Cameroon-Air-Quality-Prediction.git",
        entrypoint="src/data/data_pipeline.py:main_flow",
    ).deploy(
        name="Air-Quality-Pipeline-Deployment",
        work_pool_name="my-ecs-pool",
        image=DockerImage(
            name="prefect-flows:latest",
            platform="linux/amd64"
        ),
        schedules=[
            CronSchedule(
                cron="18 12 * * *",
                timezone="Africa/Douala"
            )
        ]
    )

### Version info

```Text
Here is the output of 'prefect version' : 

Version:             3.1.0
API version:         0.8.4
Python version:      3.10.12
Git commit:          a83ba39b
Built:               Thu, Oct 31, 2024 12:43 PM
OS/Arch:             linux/x86_64
Profile:             local
Server type:         cloud
Pydantic version:    2.9.2
Integrations:
  prefect-github:    0.3.0
  prefect-aws:       0.5.0
  prefect-docker:    0.6.2
```

### Additional context

Work Pool: I’m using "my-ecs-pool", which is an ECS push work pool automatically provisioned using Prefect. I followed [this documentation](https://docs.prefect.io/v3/deploy/infrastructure-examples/serverless) to set it up.

The error message suggests the issue may be related to a mismatch in platform or architecture when running the Docker image. However, I’ve specified platform="linux/amd64" in the deployment.

**Comment by zzstoatzz** [comment]
hi @NembotJules - this error
```
This may be caused by attempting to run an image with a misspecified platform or architecture.
```

is a common gotcha related to the platform on which you build your docker images. Can you produce a minimal reproduction that shows this error is related to the implementation of Prefect?

**Reply by NembotJules** [reply]
Hi @zzstoatzz, I’m not entirely sure if this issue is related to Prefect's implementation itself, as this is my first attempt at deploying a flow. My platform architecture is x86_64, so I’ve specified platform="linux/amd64" in the code I shared. I would greatly appreciate it if you could offer some guidance on how to resolve this issue.

**Reply by zzstoatzz** [reply]
hrrm a couple thoughts:

I think `"prefect-flows:latest"` should contain a registry name, like
- ecr: `arn:aws:ecr:us-east-1:<account-id>:repository/<repo1>`
- dockerhub: `mydockerhubusername/myrepo:mytag`

can you confirm that your image was pushed to the place you expect at deployment time? i.e. upon running the python file containing your `.deploy()` call

I suspect it may have been saved as a local image wrt the machine that created the deployment, and then the remote infra could not pull it.

otherwise in containerized situations, I often recommend starting with a local docker worker because it gives a sense of the moving pieces:

- run a worker (`--install-policy` is just shorthand for `pip install prefect-docker` beforehand)
```
prefect worker start --pool docker-test --type-docker --install-policy always
```

- point your deployment at that by setting `work_pool_name='docker-test'` in your `.deploy()` call

because then you can just watch the `prefect worker start` process and see if something funky is going on.

if videos are a format you find helpful, I go over workers and work pools [here](https://www.youtube.com/watch?v=KzumEnsLOe0)

**Reply by NembotJules** [reply]
@zzstoatzz  It still doesn't work. I tried deploying my script using Prefect's managed pool, but when I run it, the flow crashes with the following error:
"Failed due to a(n) crashed flow run - Flow run process exited with non-zero status code 1." Further investigation is needed to determine the cause of the crash.

**Reply by NembotJules** [reply]
@zzstoatzz even deploying and running a simple "Hello, World!" function with the Prefect managed work pool fails. The tool is really easy to get started with, but the sheer number of bugs completely negates that advantage.

**Reply by zzstoatzz** [reply, marked as answer]
> Failed due to a(n) crashed flow run - Flow run process exited with non-zero status code 1.

this is not necessarily a bug - this is an error that could be more informative but is a common error among new users when you:
- forget to install 3rd party deps into your image that your code requires (the prefect base image is default)
- don't provide credentials to pull your code from a private repo

did you follow along with the video I linked? 

as I mention, we could certainly improve that error message, but concrete suggestions and MREs are most useful to maintainers

**Reply by NembotJules** [reply]
@zzstoatzz  I’m extremely sorry, the error I faced was entirely my fault. I installed all the dependencies in my image, and it’s now working. Thank you for your hard work and for always taking the time to answer my questions.

**Reply by zzstoatzz** [reply]
> @zzstoatzz  I’m extremely sorry, the error I faced was entirely my fault. I installed all the dependencies in my image, and it’s now working. Thank you for your hard work and for always taking the time to answer my questions.

thank you for following up! glad you got it working
